File: app/src/main/java/com/p1/uber/receivers/CancelReceiver.java
package com.p1.uber.receivers;

import android.app.NotificationManager;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.p1.uber.provides.ClientBookingProvider;

public class CancelReceiver extends BroadcastReceiver {

    private ClientBookingProvider mClientBookingProvider;


    @Override
    public void onReceive(Context context, Intent intent) {

        String IdClient = intent.getExtras().getString("IdClient");
        mClientBookingProvider = new ClientBookingProvider();
        mClientBookingProvider.updateStatus(IdClient, "cancel");

        NotificationManager manager = (NotificationManager) context.getSystemService(context.NOTIFICATION_SERVICE);
        manager.cancel(2);

    }

}

File: app/src/main/java/com/p1/uberfares/receivers/CancelReceiver.kt
package com.p1.uberfares.receivers

import android.app.NotificationManager
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import androidx.annotation.RequiresApi
import com.p1.uberfares.provides.ClientBookingProvider

class CancelReceiver : BroadcastReceiver() {
    private var mClientBookingProvider: ClientBookingProvider? = null
    @RequiresApi(Build.VERSION_CODES.O)
    override fun onReceive(context: Context, intent: Intent) {
        val sharedPreference = context.getSharedPreferences("Clients", Context.MODE_PRIVATE)

        var IdClient = sharedPreference.getString("IdClient", "")
        mClientBookingProvider = ClientBookingProvider()
        mClientBookingProvider!!.updateStatus(IdClient, "cancel")
        val manager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        manager.cancel(2)
    }
}